libs/log/sem_logger.go

```diff
@@ -0,0 +1,176 @@
+package log
+
+// Selective Entity Monitor (SEM) Logger enables rule based log traces
+
+type SemLogger interface {
+	Logger
+	Entry(SemEntity, []byte)
+	Exit(SemEntity, []byte)
+	AddRule(SemEntity, []byte)
+	DeleteAll()
+	Status() (SemStatus, error)
+}
+
+// type loglevel byte
+type SemEntity int
+
+const (
+	SemTransaction SemEntity = iota + 1
+	SemAddress
+)
+
+type SemStatus struct {
+	Active bool
+	Rules  map[SemEntity]map[string]bool
+}
+
+type semLogger struct {
+	logBackend Logger
+	rules      map[SemEntity]map[string]uint
+	active     bool
+	enabled    bool
+}
+
+func (l *semLogger) Error(msg string, kvals ...interface{}) {
+	l.logBackend.Error(msg, kvals...)
+}
+
+func (l *semLogger) Info(msg string, kvals ...interface{}) {
+	if !l.enabled || l.active {
+		l.logBackend.Info(msg, kvals...)
+	}
+}
+
+func (l *semLogger) Debug(msg string, kvals ...interface{}) {
+	if !l.enabled || l.active {
+		l.logBackend.Debug(msg, kvals...)
+	}
+}
+
+func (l *semLogger) With(keyvals ...interface{}) Logger {
+	sl := new(semLogger)
+	sl.logBackend = l.logBackend.With(keyvals...)
+	sl.rules = l.rules
```

Comment:
```suggestion
	sl := &semLogger{
		logBackend: l.logBackend.With(keyvals...),
		rules: l.rules
	}
```
A suggestion, up to you